src/backend/rest.ts

```diff
@@ -72,10 +72,19 @@ export default class RestApiBackend implements Backend {
       url: '/version',
       method: 'GET',
     });
-    // TODO: check if this version matches the edition!
+    const checkMkVersion: string = result.data.versions.checkmk;
+    if (checkMkVersion.startsWith("2.0") || checkMkVersion.startsWith("1.6")) {
+      throw new Error(`CheckMK version ${checkMkVersion} is not supported for this plugin, but you can set the backend to the '< 2.2' version and use at your own risk.`)
+    }
+    if (checkMkVersion.startsWith("2.1")) {
+      throw new Error(`CheckMK version ${checkMkVersion} does not support querying graph data using this backend. Please set the backend option to '< 2.2' and make sure to have the Web Api enabled.`)
+    }
+    if (this.datasource.getEdition() === "CEE" && result.data.edition === "raw") {
+      throw new Error("This plugin is configured to use the Enterprise Edition, but the server is a RAW edition. Please check the configuration of this plugin.")
```

Comment:
i think we can be me more specific here:
```suggestion
      throw new Error("The data source specified the Checkmk Enterprise Edition, but Checkmk Raw Edition was detected. Please choose the raw edition in the data source settings.")
```